### PR TITLE
🛡 Revendor `gardener/gardener@v1.39`, update Golang version to `1.17`, enable `ServiceAccount` token projection

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:
* Revendor `gardener/gardener@v1.39`.
* Update Golang version to `1.17`.
* When the extension is running on a seed with a gardenlet of at least `v1.37` then
  * `ServiceAccount` token projection is enabled.
  * the `cilium`-related `Pod`s deployed into shoot clusters
    * uses `ServiceAccount` token projection.

ℹ️ The `branch.cfg` definition is still using Golang `1.16`, hence, the `verify` step fails. Once this PR is approved, we can adapt the `branch.cfg` (also update to `1.17`) and then re-trigger the CI pipeline for this PR.

**Which issue(s) this PR fixes**:
Part of gardener/gardener#4659
Part of gardener/gardener#4878

**Special notes for your reviewer**:
✅ ~Depends on https://github.com/gardener/gardener/pull/5162 and `v1.39` release, hence, PR is in draft state.~

**Release note**:

```noteworthy developer
The Golang version has been updated to `1.17.5`.
```
```noteworthy operator
The extension controller uses a projected `ServiceAccount` token in case it runs on a seed with a gardenlet of at least `v1.37` or higher. The same applies for the `cilium`-related `Pod`s deployed to shoot cluster.
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-FROM eu.gcr.io/gardener-project/3rd/golang:1.16.2 AS builder
+FROM golang:1.17.5 AS builder
 
 WORKDIR /go/src/github.com/gardener/gardener-extension-networking-cilium
 COPY . .
 RUN make install
 
 ############# gardener-extension-networking-cilium
-FROM eu.gcr.io/gardener-project/3rd/alpine:3.13.2 AS gardener-extension-networking-cilium
+FROM alpine:1.15.0 AS gardener-extension-networking-cilium
```
Comment (DockToFuture) [inline Dockerfile:9]:
```suggestion
FROM alpine:3.15.0 AS gardener-extension-networking-cilium
```

**Comment by DockToFuture** [comment]
When trying to build the image it fails with
```
#10 31.92 pkg/controller/actuator_delete.go:30:33: not enough arguments in call to applyMonitoringConfig
#10 31.92       have (context.Context, client.Client, "github.com/gardener/gardener/pkg/client/kubernetes".ChartApplier, *"github.com/gardener/gardener/pkg/apis/extensions/v1alpha1".Network, bool)
#10 31.92 	want (context.Context, client.Client, "github.com/gardener/gardener/pkg/client/kubernetes".ChartApplier, *"github.com/gardener/gardener/pkg/apis/extensions/v1alpha1".Network, bool, bool)
```

There is a bool value missing.

**Comment by rfranzke** [comment]
@DockToFuture fixed

**Review by DockToFuture — APPROVED** [review]
/lgtm, PR was tested locally.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Comment by timuthy** [comment]
/unassign
PR was reviewed and tested by repo maintainers.

**Pull request merged**